src/DataGrid.tsx

```diff
@@ -423,13 +441,34 @@ function DataGrid<R, K extends keyof R, SR>({
   }
 
   function handleCellInput(event: React.KeyboardEvent<HTMLDivElement>) {
+    if (!isCellWithinBounds(selectedPosition)) return;
     const { key } = event;
     const column = columns[selectedPosition.idx];
     const row = rows[selectedPosition.rowIdx];
-    const canOpenEditor = selectedPosition.mode === 'SELECT' && isCellEditable(selectedPosition);
-    const isActivatedByUser = (column.unsafe_onCellInput ?? legacyCellInput)(event, row) === true;
+    const isUsingEditor2 = column.editor2 !== undefined;
 
-    if (canOpenEditor && (key === 'Enter' || isActivatedByUser)) {
+    if (selectedPosition.mode === 'EDIT') {
+      if (isUsingEditor2 && (key === 'Enter' || key === 'F2')) {
```

Comment:
I think it is not possible to have one solution for every type of editor and we need some way to to tweak the default behavior. If the editor do not need to commit changes on `Enter` then it should handle the `keydown` event and stop propagation. The question is what default behavior we want to use? Right now we do the following 
- Open editor on `Enter`, `F2` and alphanumeric keys
- Cancel changes on `Escape`
- Commit changes on `Enter` and `F2`
- Commit changes when another cell is clicked
- Commit changes when header or summary row is clicked

Any suggestions?

Reply:
I think for the `Enter` case, we shall leave a prop for the user to decide if we want to use `Enter` to close the the editor.
It is a common use case that people might want to put a save button with 2 inputs for example, `Enter` would mean something else for them. As for F2, maybe we should keep like excel to only open the cell rather than closing it. `Esc` is pretty straightforward, just close without saving / commit.